File: android/app/src/main/java/com/secondbrain/domain/model/Person.kt
package com.secondbrain.domain.model

data class Person(
    val id: String,
    val type: String = "person",
    val name: String,
    val status: String = "active",
    val contacts: List<Contact> = emptyList(),
    val socialLinks: List<SocialLink> = emptyList(),
    val tags: List<String> = emptyList(),
    val links: List<String> = emptyList(),
    val notes: String = "",
    val body: String = "",
    val createdAt: String = "",
    val updatedAt: String = ""
)

data class Contact(
    val type: String,  // phone, email, social
    val value: String,
    val label: String = ""
)

data class SocialLink(
    val platform: String,
    val url: String
)

File: backend/internal/model/person.go
package model

import "time"

type Person struct {
	BaseEntity  `yaml:",inline" json:",inline"`
	Name        string       `yaml:"name" json:"name"`
	Contacts    []Contact    `yaml:"contacts,omitempty" json:"contacts,omitempty"`
	SocialLinks []SocialLink `yaml:"social_links,omitempty" json:"social_links,omitempty"`
	Notes       string       `yaml:"notes,omitempty" json:"notes,omitempty"`
	Body        string       `yaml:"-" json:"body"`
}

func (p *Person) SetBody(body string) { p.Body = body }

func NewPerson(id, name string) *Person {
	now := time.Now().UTC()
	return &Person{
		BaseEntity: BaseEntity{
			ID:        id,
			Type:      TypePerson,
			Status:    StatusActive,
			Tags:      []string{},
			Links:     []string{},
			CreatedAt: now,
			UpdatedAt: now,
		},
		Name:        name,
		Contacts:    []Contact{},
		SocialLinks: []SocialLink{},
		Notes:       "",
		Body:        "",
	}
}

func (p *Person) Validate() error {
	if err := ValidateBase(&p.BaseEntity); err != nil {
		return err
	}
	if p.Name == "" {
		return NewValidationError("person.name is required")
	}
	return nil
}
